sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1010,7 +1014,16 @@ class Analyzer(
                 s"(valid range is [1, ${aggs.size}])")
           case o => o
         }
-        Aggregate(newGroups, aggs, child)
+        var filterGroups: Seq[Expression] = Nil
+        for (ng <- newGroups)
+        if (ng.isInstanceOf[Alias]) {
```

Comment:
We can use pattern matching here like:

ng match {
    case a: Alias => if (!isIntLiteral(a.child)) filterGroups :+= ng
    case  _ => filterGroups :+= ng
}